src/main/java/fi/aalto/cs/apluscourses/ui/courseproject/CourseProjectActionDialogs.java

```diff
@@ -1,9 +1,9 @@
-package fi.aalto.cs.apluscourses.ui.base;
+package fi.aalto.cs.apluscourses.ui.courseproject;
 
+import com.intellij.openapi.project.Project;
 import org.jetbrains.annotations.NotNull;
 
-public interface Dialogs {
-  void showInformationDialog(@NotNull String message, @NotNull String title);
+public interface CourseProjectActionDialogs {
```

Comment:
as with most programming challenges, one can relatively easily find good arguments backed up [by some blogpost](https://octoperf.com/blog/2016/10/27/impl-classes-are-evil/) to support its' points. but it's not my point here :smile: 

I'd like to draw attention to a higher-level matter: in several places, we have been utilizing the approach of `Interface` + `Impl` class. this approach has certain benefits in certain cases because of (... blablabla ..., just google). thus, I'd like us to question if we are really the subject of those cases? 

as @OlliKiljunen already mentioned once, there is [YAGNI principle](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it) and it makes a lot of sense to me. Now the actual questions (I have no answers myself): 
+ what is the benefit of having an `Interface` here? 
+ would that make more sense to create a concrete `Class` in this case and introduce an `Interface` only when there's a clear need to use `Dialogs` elsewhere?
+ is it already known, that these `Dialogs` will be reused (have another implementation)?
+ a class just moved from `Dialogs` to become `CourseProjectActionDialogs`, could this be a sign, that this is now a conceptually same entity (`Interface` + `Imlp`)?

@nikke234 @OlliKiljunen :grey_question:

Reply:
and yes, maybe I just don't know how to cook them (`Interfaces`) properly :stuck_out_tongue_closed_eyes:

Reply:
Interesting blog post :+1: As Olli mentioned, the main reason for the separation here is unit-testability. I do agree that the `Impl` naming is somewhat annoying, but I think the benefits of testability outweigh the negatives here :smile: In this case I removed the original more generic `Dialogs` interface which could be used elsewhere as well, we could consider having a generic `Dialogs` interface again.